webapp/packages/plugin-datasource-transaction-manager/src/TransactionLogs/useTransactionLogs.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { observable } from 'mobx';
+
+import { useObservableRef } from '@cloudbeaver/core-blocks';
+import type { ConnectionExecutionContext } from '@cloudbeaver/core-connections';
+import type { TransactionLogInfoItem } from '@cloudbeaver/core-sdk';
+import type { ILoadableState } from '@cloudbeaver/core-utils';
+
+interface Payload {
+  transaction: ConnectionExecutionContext;
+}
+
+interface State extends ILoadableState {
+  logs: TransactionLogInfoItem[];
+  exception: Error | null;
+  promise: Promise<TransactionLogInfoItem[] | undefined> | null;
+  payload: Payload;
+}
+
+export function useTransactionLogs(payload: Payload) {
+  const state = useObservableRef<State>(
+    () => ({
+      logs: [],
+      exception: null,
+      promise: null,
+      isLoaded() {
+        return this.logs.length > 0;
+      },
```

Comment:
This is incorrect cuz the backend may return empty logs, and in that case, the state must be isLoaded == true
you can use different initial states that don't have collisions with data returned by `payload.transaction.getLogs();` or a different flag to store the state